config/environment.rb

```diff
@@ -1,3 +1,19 @@
+# Strip whitespace from environment values for proper API to app handshake.
+class AppEnv
+  def self.[](key)
+    ENV.key?(key) && ENV[key] = ENV[key].strip
+  end
+end
+
+AppEnv['WEATHER_API_KEY']
+AppEnv['SLACK_CLIENT_ID']
+AppEnv['SLACK_API_SECRET']
+AppEnv['SLACK_VERIFICATION_TOKEN']
+AppEnv['TWITTER_CONSUMER_KEY']
+AppEnv['TWITTER_CONSUMER_SECRET']
+AppEnv['TWITTER_ACCESS_TOKEN']
+AppEnv['TWITTER_ACCESS_SECRET']
```

Comment:
Sorry I wasn't clear. I was thinking we'd avoid mutating the ENV variables directly:
```
class AppEnv
  def self.[](key)
    ENV[key].strip
  end
end
```
and then find instances of `ENV[...]` in the `app` directory to replace them with `AppEnv[...]`

I think we should also put AppEnv in its own file in `lib/app_env.rb`

Reply:
What instances would leading spaces be necessary for some rails variables? If we know these instances, would it be less costly to overwrite specific variables in the ENV file once so the application doesn't have to keep parsing the white spaces every time we start the app?

Reply:
.strip is O(n) where n is the length of the string. I think that because the length of the strings that need to be stripped are not long (around 25) and not variable in length, it is probably not a bottleneck in the runtime of the application. What are everyone's thoughts on this?

Reply:
Looks like `n` is the length of the leading and trailing whitespace (https://ruby-doc.org/core-3.0.1/String.html#method-i-strip / https://bugs.ruby-lang.org/issues/12788), but either way isn't a problem here.  If you're concerned about `String#strip` performance, you're already using the wrong language.  ;)

Reply:
I don't think we should mutate the existing ENV hash since it would be difficult to discover and troubleshoot *if* it is the ever the wrong operation.

I would instead opt for adding a new class:
```
class AppEnv
  def self.[](key)
    ENV[key].strip
  end
end
```
and changing existing usages of `ENV` found in our `app/` folder. An example:
`app/controllers/web_hooks/github_controller.rb`
```
 35   private def webhook_secret(_payload)
 36     ENV['GITHUB_WEBHOOK_SECRET']
 37   end
```
would become
```
 35   private def webhook_secret(_payload)
 36     AppEnv['GITHUB_WEBHOOK_SECRET']
 37   end
 ```

Why this change instead of the original?
* Set up other developers to do the right thing. Some time in the future when another developer wants to add a new ENV variable for some new service, they're probably going to look in the application for other services that do the same thing and copy it. With `AppEnv` used in our app for all of our env vars, they will likely copy that. They might miss the allow/disallow list of env variables to mutate in ENV.
* Discoverability. If someone needs to find out why whitespace is being stripped out, `AppEnv` in their call provides a clear callstack to follow. Trying to find a mutation in `config/environment.rb` is harder.

I think these are good coding practices to follow, even if the likelihood of us or rails ever needing whitespace in our ENV vars is pretty low.